tinker/faculty_bios/faculty_bio_controller.py

```diff
@@ -122,21 +124,201 @@ def _iterate_child_xml(self, child, author):
+                        'job_title' + str(max_jobs): title
+                    }
+                    # Adds the dictionary to the list
+                    my_list.append(my_dict)
+
+                # max_edu will hold the maximum number of jobs someone has
+                max_edu = 0
+                for edu in soup.findAll('education'):
+                    max_edu += 1
+                    # Creates the dictionary with the relevant information
+                    my_dict = {
+                        'school-edu' + str(max_edu): edu.find('school').text,
+                        'degree-earned' + str(max_edu): edu.find('degree-earned').text,
+                        'year' + str(max_edu): edu.find('year').text,
+                    }
+                    # Adds the dictionary to the list
+                    my_list.append(my_dict)
+
+                # Creates a dictionary with number of jobs everyone has
+                max_dict = {
+                    'max-jobs': max_jobs,
+                    'max-edu': max_edu
+                }
+
+                # Formats these as wysiwyg's since staff/faculty can add formatting to these
+                biography = ""
+                if soup.find('biography').text:
+                    biography = self.strip_html_tags(unicode(soup.find('biography').renderContents('utf-8', True), "utf-8"))
+                courses = ""
+                if soup.find('courses').text:
+                    courses = self.strip_html_tags(unicode(soup.find('courses').renderContents('utf-8', True), "utf-8"))
+                awards = ""
+                if soup.find('awards').text:
+                    awards = self.strip_html_tags(unicode(soup.find('awards').renderContents('utf-8', True), "utf-8"))
+                publications = ""
+                if soup.find('publications').text:
+                    publications = self.strip_html_tags(unicode(soup.find('publications').renderContents('utf-8', True), "utf-8"))
+                presentations = ""
+                if soup.find('presentations').text:
+                    presentations = self.strip_html_tags(unicode(soup.find('publications').renderContents('utf-8', True), "utf-8"))
+                certificates = ""
+                if soup.find('certificates').text:
+                    certificates = self.strip_html_tags(unicode(soup.find('certificates').renderContents('utf-8', True), "utf-8"))
+                organizations = ""
+                if soup.find('organizations').text:
+                    organizations = self.strip_html_tags(unicode(soup.find('organizations').renderContents('utf-8', True), "utf-8"))
+                hobbies = ""
+                if soup.find('hobbies').text:
+                    hobbies = self.strip_html_tags(unicode(soup.find('hobbies').renderContents('utf-8', True), "utf-8"))
+
+                # This is the initial dictionary of what will be returned
+                page_values = {
```

Comment:
Why not either build this as you go, or do this same idea for the job titles/education? This would allow you to read from everything at the end?

Reply:
Done